.github/workflows/deploy_qa.yml

```diff
@@ -9,9 +9,7 @@ on:
   pull_request:
     branches:
       - release
-    types:
-      - opened
-  workflow_dispatch: # This enables manual triggers
+    types: [opened, edited, synchronize] # PR opened, edited(PR title, etc), new commits are pushed.
 
```

Comment:
workflow_dispatch doesn't work here due to lacking reference to a PR

      - name: Extract Version from PR Number
        id: extract_version
        run: |
          VERSION=${{ github.event.pull_request.number }} 
          echo "VERSION=$VERSION" >> $GITHUB_ENV
          echo "::set-output name=VERSION::$VERSION"
          echo "Extracted Version: $VERSION"